### ISSUE TITLE
tflite can't ResizeInputTensor size

### ISSUE THREAD

**Issue by TPcoding** [tensorflow/tensorflow#22377]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:N/A
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:16.04.1-Ubuntu
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:N/A
- **TensorFlow installed from (source or binary)**:source
- **TensorFlow version (use command below)**:1.11.0rc0
- **Python version**:3.5
- **Bazel version (if compiling from source)**:0.16.1
- **GCC/Compiler version (if compiling from source)**: (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
- **CUDA/cuDNN version**:N/A
- **GPU model and memory**:N/A
- **Exact command to reproduce**:See below
### Describe the problem
When I use tflite,I set ResizeInputTensor different from model, error show:
```
tensorflow/contrib/lite/kernels/reshape.cc:58 num_input_elements != num_output_elements (880 != 440)
Node number 0 (RESHAPE) failed to prepare.
```
fisrt,my net input is [?,440], I convert it by set intput to [1,440]. I want to use it in forward with different size [n,440].Maybe I conert it to tflite wrong?
### Source code / logs
*my code:
```
int main() {

    // load model
    std::unique_ptr<tflite::FlatBufferModel> model;
    std::unique_ptr<tflite::Interpreter> interpreter;
    model = tflite::FlatBufferModel::BuildFromFile("./moble.tflite");
    if (!model) {
	cout << "load model err." << endl;
	exit(-1);
    }

    // build op
    tflite::ops::builtin::BuiltinOpResolver resolver;

    tflite::InterpreterBuilder (*model, resolver)(&interpreter);
    if (!interpreter) {
	cout << "Failed to construct interpreter\n" << endl;
	exit(-1);
    }

    interpreter->UseNNAPI(0);
    interpreter->SetNumThreads(1);

    // allocate tensors space
    int input = interpreter->inputs()[0];
    interpreter->ResizeInputTensor(input, {2,440});
    if (interpreter->AllocateTensors() != kTfLiteOk) {
	cout << "Failed to allocate tensors.\n";
	exit(-1);
    }

    // set input
    float *in = interpreter->typed_tensor<float>(input);
    for (int i = 0; i < 440 * 2; i++) {
	in[i] = i + 1.0f;
    }

    // inference
    if (interpreter->Invoke() != kTfLiteOk) {
	cout << "Failed to invoke!\n";
	exit(-1);
    }

    // get result
    float *output = interpreter->typed_output_tensor<float>(0);
    for (int i = 0; i < 200; i++) {
	cout << output[i] << " ";
    }
    cout << endl;
    return 0;
}
```
* python convert script
** my net
```
bazel run tensorflow/tools/graph_transforms:summarize_graph -- --in_graph=/home/aisp/work_mirror/tensorflow/model/cnn/vocal_print_model/graph_vprint.pb
```
result
```
Found 1 possible inputs: (name=ac_input, type=float(1), shape=[?,440]) 
No variables spotted.
Found 1 possible outputs: (name=Inference/final_output/output, op=Identity) 
Found 382867 (382.87k) const parameters, 0 (0) variable parameters, and 0 control_edges
Op types used: 50 Const, 13 Reshape, 13 Identity, 12 Transpose, 6 BiasAdd, 4 Conv2D, 4 Relu, 3 ConcatV2, 3 Mul, 2 GatherV2, 2 Add, 2 Cast, 2 Prod, 2 Mean, 2 MaxPool, 2 MatMul, 1 Pack, 1 Placeholder, 1 Range, 1 ListDiff, 1 Less, 1 Shape, 1 Sqrt, 1 Sub, 1 GreaterEqual
```
** transform net
```
bazel run tensorflow/tools/graph_transforms:transform_graph -- --in_graph=/home/aisp/work_mirror/tensorflow/model/cnn/vocal_print_model/graph_vprint.pb --out_graph=/home/aisp/work_mirror/tensorflow/model/cnn/vocal_print_model/transformed_graph_simple.pb --inputs='ac_input' --outputs='Inference/final_output/output' --transforms='strip_unused_nodes(type=float,shape="1,440") fold_constants(ingore_errors=true) fold_batch_norms fold_old_batch_norms'
```
** convert to lite
```
tflite_convert   --output_file=$(pwd)/model/cnn/vocal_print_model/moble.tflite  --graph_def_file=$(pwd)/model/cnn/vocal_print_model/transformed_graph_simple.pb   --input_arrays=ac_input   --output_arrays=Inference/final_output/output --input_shapes=1,440
```

**Comment by TPcoding** [comment]
@aselle Please help me look at this problem. It really makes me wonder.

**Comment by TPcoding** [comment]
Here are my model.

[model.tar.gz](https://github.com/tensorflow/tensorflow/files/2400554/model.tar.gz)

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by shawn-tian** [comment]
@TPcoding did you solve this issue? Thank you.

**Comment by TPcoding** [comment]
> @TPcoding did you solve this issue? Thank you.

No, I rewrite my graph ,only support one frame input.This issue still open.

**Comment by aselle** [comment]
@TPCoding, not all models can have their inputs resized for example

x = tf.placeholder()
y = tf.reshape(x, (440,1))
... do more stuff

This model needs to have shapes that have 440 elements. you cannot feed a tensor that is length 3 it would have a similar error.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 32 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by shahid14** [comment]
@aselle I had similar issue,  on tensorflow lite demo in Android, while it's resolved in IOS demo version.

Please help me resolve this issue in android as well. I h'v mentioned details of bug in official demo.
I'm not able to work around it, please let me know your response on it!!

Link to opened issue : 
https://github.com/tensorflow/tensorflow/issues/23940

**Comment by TPcoding** [comment]
@aselle  Half year passed,Does this issues will be done in new tflite version plan. I really need this function.Recently, I have to implement several model which has variable sequence length when input to tflite.

**Comment by kmader** [comment]
I am having the same issue with a fully convolution 1D network (in tf.keras having variable model size is not a problem). I export the model with a fixed size of 180x3 (since it won't export with None in any dimension) and then when I try to change the size with ResizeInput I get the message above.

**Comment by melihcan06** [comment]
when i get this error = num_output elements != num_output elements [4004] != [1001] i changed input dimensions.from [1][200][200][3] to [1][128][128][3] . it works.I think model need this dimensions.I get model dimensions from pbtxt file.I create pbtxt file from= .....................................................
from keras.layers import Dense
import numpy
from keras.models import load_model
from tensorflow.python.tools import freeze_graph, optimize_for_inference_lib
from keras import backend as K
from sklearn.model_selection import train_test_split
from keras.wrappers.scikit_learn import KerasRegressor
import tensorflow as tf

seed = 7
numpy.random.seed(seed)


model=load_model("ResNet50")



def export_model(saver, model, input_node_names, output_node_name,MODEL_NAME):
    tf.train.write_graph(K.get_session().graph_def, 'out', \
                         MODEL_NAME + '_graph.pbtxt')

    saver.save(K.get_session(), 'out/' + MODEL_NAME + '.chkp')

    freeze_graph.freeze_graph('out/' + MODEL_NAME + '_graph.pbtxt', None, \
                              False, 'out/' + MODEL_NAME + '.chkp', output_node_name, \
                              "save/restore_all", "save/Const:0", \
                              'out/modelim' + MODEL_NAME + '.pb', True, "")

    input_graph_def = tf.GraphDef()
    with tf.gfile.Open('out/modelim' + MODEL_NAME + '.pb', "rb") as f:
        input_graph_def.ParseFromString(f.read())

    output_graph_def = optimize_for_inference_lib.optimize_for_inference(
        input_graph_def, input_node_names, [output_node_name],
        tf.float32.as_datatype_enum)

    with tf.gfile.FastGFile('out/opt_' + MODEL_NAME + '.pb', "wb") as f:
        f.write(output_graph_def.SerializeToString())

    print("graph saved!")

    return

estimators = []
estimator = KerasRegressor(build_fn=model, epochs=1, batch_size=50, verbose=2)
inputName=model.input.name[:-2]
outPutName=model.output.name[:-2]

with open("inoutname.txt","w") as ds:
    ds.write(inputName+" "+outPutName)

export_model(tf.train.Saver(), estimator, [inputName], outPutName,"test")

**Comment by mohantym** [comment]
Hi @TPcoding !
We are checking to see if you still need help on this issue. We recommend that you upgrade to 2.6 which is latest stable version of TF and let us know if the issue still persists in newer versions. Thanks!

**Comment by SefaZeng** [comment]
Any solutions? Meeting the same issue.

**Comment by ArchieNine** [comment]
Any solutions updated? same problem
